ddl/util/syncer_test.go

```diff
@@ -0,0 +1,240 @@
+	globalVer, err := d.SchemaSyncer().MustGetGlobalVersion(ctx)
+	if err != nil {
+		t.Fatalf("client get global version failed %v", err)
+	}
+	if InitialVersion != fmt.Sprintf("%d", globalVer) {
+		t.Fatalf("client get global version %d isn't equal to init version %s", globalVer, InitialVersion)
+	}
+	childCtx, _ := goctx.WithTimeout(ctx, minInterval)
+	_, err = d.SchemaSyncer().MustGetGlobalVersion(childCtx)
+	if !isTimeoutError(err) {
+		t.Fatalf("client get global version result not match, err %v", err)
+	}
+
+	d1 := NewDDL(ctx, cli, store, nil, nil, testLease, nil)
+	defer d1.Stop()
+	if err = d1.SchemaSyncer().Init(ctx); err != nil {
+		t.Fatalf("schema version syncer init failed %v", err)
+	}
+
+	// for watchCh
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	currentVer := int64(123)
+	go func() {
+		defer wg.Done()
+		select {
+		case resp := <-d.SchemaSyncer().GlobalVersionCh():
+			if len(resp.Events) < 1 {
+				t.Fatalf("get chan events count less than 1")
+			}
+			checkRespKV(t, 1, DDLGlobalSchemaVersion, fmt.Sprintf("%v", currentVer), resp.Events[0].Kv)
+		case <-time.After(100 * time.Millisecond):
+			t.Fatalf("get udpate version failed")
+		}
+	}()
+
+	// for update latestSchemaVersion
+	err = d.SchemaSyncer().OwnerUpdateGlobalVersion(ctx, currentVer)
+	if err != nil {
+		t.Fatalf("update latest schema version failed %v", err)
+	}
+
+	wg.Wait()
+
+	// for CheckAllVersions
+	childCtx, cancel := goctx.WithTimeout(ctx, 20*time.Millisecond)
+	err = d.SchemaSyncer().OwnerCheckAllVersions(childCtx, currentVer)
+	if err == nil {
+		t.Fatalf("check result not match")
+	}
+	cancel()
+
+	// for UpdateSelfVersion
+	childCtx, cancel = goctx.WithTimeout(ctx, 100*time.Millisecond)
+	err = d.SchemaSyncer().UpdateSelfVersion(childCtx, currentVer)
+	if err != nil {
+		t.Fatalf("update self version failed %v", errors.ErrorStack(err))
+	}
+	cancel()
+	childCtx, cancel = goctx.WithTimeout(ctx, 100*time.Millisecond)
+	err = d1.SchemaSyncer().UpdateSelfVersion(childCtx, currentVer)
+	if err != nil {
+		t.Fatalf("update self version failed %v", errors.ErrorStack(err))
+	}
+	cancel()
+	childCtx, _ = goctx.WithTimeout(ctx, minInterval)
+	err = d1.SchemaSyncer().UpdateSelfVersion(childCtx, currentVer)
+	if !isTimeoutError(err) {
+		t.Fatalf("update self version result not match, err %v", err)
+	}
+
+	// for CheckAllVersions
+	childCtx, _ = goctx.WithTimeout(ctx, 100*time.Millisecond)
+	err = d.SchemaSyncer().OwnerCheckAllVersions(childCtx, currentVer-1)
+	if err != nil {
+		t.Fatalf("check all versions failed %v", err)
+	}
+	childCtx, _ = goctx.WithTimeout(ctx, 100*time.Millisecond)
+	err = d.SchemaSyncer().OwnerCheckAllVersions(childCtx, currentVer)
+	if err != nil {
+		t.Fatalf("check all versions failed %v", err)
+	}
+	childCtx, _ = goctx.WithTimeout(ctx, minInterval)
+	err = d.SchemaSyncer().OwnerCheckAllVersions(childCtx, currentVer)
+	if !isTimeoutError(err) {
+		t.Fatalf("check all versions result not match, err %v", err)
+	}
+
+	// for StartCleanWork
```

Comment:
This test case is added in this PR, others are moved from `ddl_etcd_test/unit_test/syncer_test.go`.